contracts/Marketplace.sol

```diff
@@ -103,7 +107,7 @@ contract Marketplace is Proofs, StateRetrieval, Endian {
     _marketplaceTotals.received += amount;
     _transferFrom(msg.sender, amount);
 
-    emit StorageRequested(id, request.ask, request.expiry);
+    emit StorageRequested(id, request.ask, _requestContexts[id].expiresAt);
```

Comment:
I think this is definitely the way to go. The node will need to consume this value and set the local timer accordingly instead of calculating the value. The pre-calculation of `expiry` on the node end was possibly why a timestamp was initially used for expiry. 

We could probably update the `purchasing/states/pending` state to listen for the `StorageRequested` event before going to the `submitted` state, passing in the `expiresAt` value from the event to the `submitted` state. In the `submitted` state, we can use `await clock.waitUntil(expiresAt)` and cancel that future if the request is fulfilled.